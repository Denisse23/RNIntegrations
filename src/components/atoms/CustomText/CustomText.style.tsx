import styled from '@emotion/native';
import { CustomTextProps } from './CustomText';

export const StyledText = styled.Text<CustomTextProps>(
  ({ theme, size = 'sm', type = 'regular' }) => ({
    color: theme.colors.text300,
    fontSize: theme.fontSizes[size],
    fontWeight: type === 'bold' ? '700' : '400',
  }),
);
