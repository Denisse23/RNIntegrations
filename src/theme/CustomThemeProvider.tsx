import React from 'react';
import { useColorScheme } from 'react-native';
import { ThemeProvider } from '@emotion/react';
import { ColorScheme, DarkTheme, LightTheme } from './Themes';

const CustomThemeProvider = ({ children }: { children: React.ReactNode }) => {
  const scheme = useColorScheme();

  return (
    <ThemeProvider theme={scheme === ColorScheme.DARK ? DarkTheme : LightTheme}>
      {children}
    </ThemeProvider>
  );
};

export default CustomThemeProvider;
