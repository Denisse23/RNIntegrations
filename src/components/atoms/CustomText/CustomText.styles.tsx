import styled from '@emotion/native';
import { CustomTextProps } from './CustomText';

export const StyledText = styled.Text<CustomTextProps>(
  ({ theme, size, type }) => ({
    color: theme.colors.text300,
    fontSize: size ? theme.fontSizes[size] : theme.fontSizes.sm,
    fontWeight: type === 'bold' ? '700' : '400',
  }),
);
