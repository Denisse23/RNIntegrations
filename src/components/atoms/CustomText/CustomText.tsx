import React from 'react';
import { TextProps } from 'react-native';
import { StyledText } from './CustomText.style';
export interface CustomTextProps extends TextProps {
  size?: 'sm' | 'md' | 'lg' | 'xl' | 'xxl';
  type?: 'bold' | 'regular';
}

const CustomText = ({
  size = 'sm',
  type = 'regular',
  children,
  ...rest
}: CustomTextProps) => {
  return (
    <StyledText size={size} type={type} {...rest}>
      {children}
    </StyledText>
  );
};

export default CustomText;
