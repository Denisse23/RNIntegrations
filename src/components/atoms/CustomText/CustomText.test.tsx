import React from 'react';
import renderer from 'react-test-renderer';
import CustomText from './CustomText';
import CustomThemeProvider from '@theme/CustomThemeProvider';

describe('CustomText', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(
      <CustomThemeProvider>
        <CustomText>Test</CustomText>
      </CustomThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
