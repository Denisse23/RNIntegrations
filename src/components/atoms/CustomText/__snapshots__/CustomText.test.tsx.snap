// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomText should match the snapshot 1`] = `
<Text
  size="sm"
  style={
    Array [
      Object {
        "color": "#252536",
        "fontSize": 28.125,
        "fontWeight": "400",
      },
      undefined,
    ]
  }
  type="regular"
>
  Test
</Text>
`;
